honeybadgermpc/polynomial.py

```diff
@@ -82,6 +80,25 @@ def random(cls, degree, y0=None):
                 coeffs[0] = y0
             return cls(coeffs)
 
+        @classmethod
+        def interp_extrap(cls, xs, omega):
+            """
+            Interpolates the polynomial based on the even points omega^2i
+            then evaluates at all points omega^i
+            """
+            n = len(xs)
+            assert n & (n-1) == 0, "n must be power of 2"
+            assert pow(omega, 2*n) == 1, "omega must be 2n'th root of unity"
+            assert pow(omega, n) != 1, "omega must be primitive 2n'th root of unity"
```

Comment:
If these `assert` statements are important beyond debugging purposes, we should explicitly raise an error. A `ValueError` would probably be fine. 

Also, since this logic is reused elsewhere, it would make sense to put it in a small validation function or method, e.g.:

```python
def validate_omega(xs, omega):
    if n & (n-1) != 0:
        raise ValueError("n must be power of 2")
    if pow(omega, 2*n) != 1:
        raise ValueError("omega must be 2n'th root of unity")
    if pow(omega, n) == 1:
        raise ValueError("omega must be primitive 2n'th root of unity")
```

we could furthermore write a simple [decorator](https://docs.python.org/3/glossary.html#term-decorator).

Reply:
Also, since this logic is reused elsewhere, it would make sense to put it in a small validation function or method, e.g.:

```python
def validate_omega(xs, omega):
    if n & (n-1) != 0:
        raise ValueError("n must be power of 2")
    if pow(omega, 2*n) != 1:
        raise ValueError("omega must be 2n'th root of unity")
    if pow(omega, n) == 1:
        raise ValueError("omega must be primitive 2n'th root of unity")
```

we could furthermore write a simple [decorator](https://docs.python.org/3/glossary.html#term-decorator) if we do invoke this validation often.